server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -303,15 +310,98 @@ public void onFailure(Exception e) {
+
+        decommissionController.updateMetadataWithDecommissionStatus(DecommissionStatus.DRAINING, new ActionListener<>() {
+            @Override
+            public void onResponse(DecommissionStatus status) {
+                logger.info("updated the decommission status to [{}]", status);
+                // set the weights
+                setWeightsForAwarenessAttribute(awarenessValues, state, decommissionRequest);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error(
+                    () -> new ParameterizedMessage(
+                        "failed to update decommission status for attribute [{}] to [{}]",
+                        decommissionAttribute.toString(),
+                        DecommissionStatus.DRAINING
+                    ),
+                    e
+                );
+                // since we are not able to update the status, we will clear the voting config exclusion we have set earlier
+                clearVotingConfigExclusionAndUpdateStatus(false, false);
+            }
+        });
+    }
+
+    void setWeightsForAwarenessAttribute(List<String> awarenessValues, ClusterState state, DecommissionRequest decommissionRequest) {
+        ClusterState clusterState = clusterService.getClusterApplierService().state();
+
+        DecommissionAttributeMetadata decommissionAttributeMetadata = clusterState.metadata().decommissionAttributeMetadata();
+        assert decommissionAttributeMetadata.status().equals(DecommissionStatus.DRAINING)
+            : "unexpected status encountered while decommissioning nodes";
+        DecommissionAttribute decommissionAttribute = decommissionAttributeMetadata.decommissionAttribute();
+
+        Map<String, Double> weights = new HashMap<>();
+        awarenessValues.forEach(awarenessValue -> {
+            if (awarenessValue.equalsIgnoreCase(decommissionAttribute.attributeValue())) {
+                weights.put(awarenessValue, Double.valueOf(0.0));
+            } else {
+                weights.put(awarenessValue, Double.valueOf(1.0));
+            }
+        });
+
+        decommissionController.setWeights(decommissionAttribute.attributeName(), weights, new ActionListener<>() {
+            @Override
+            public void onResponse(ClusterPutWeightedRoutingResponse response) {
+                // Schedule the node decommission process after the weights are successfully set.
+                scheduleNodesDecommissionOnTimeout(state, decommissionRequest.getDrainingTimeout());
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                // Weights not successfully set. Fail the decommissioning process.
+                clearVotingConfigExclusionAndUpdateStatus(false, false);
+            }
+        });
+    }
+
+    void scheduleNodesDecommissionOnTimeout(ClusterState state, TimeValue timeoutForNodeDraining) {
```

Comment:
get state from applier service in this method

Reply:
Done.